bokeh/io/state.py

```diff
@@ -189,6 +192,11 @@ def output_notebook(self, notebook_type='jupyter'):
             None
 
         '''
+        if not notebook:
+            raise RuntimeError("jupyter notebook is not installed")
+        if notebook.version_info < (5, 0, 0):
+            raise RuntimeError("jupyter notebook 5.0 or higher is required")
+
```

Comment:
will this work with Jupyter Lab?

Reply:
I was about to say no, however jupyter notebook is listed as a dependency in jupyterlab's conda package and in prerequisites section in jupyterlab's README, so I presume it should work, though I will need to test this. However, what's confusing is that their minimal version number is 4.3.

Reply:
If this approach won't work, then this PR doesn't make sense, because not knowing what environment bokeh runs in, we can't have separate checks for notebook and lab.

Reply:
I think the JS approach might be safer

Reply:
Yeah, that will be better.